test/forkMainnet/RFQv2.t.sol

```diff
@@ -315,26 +437,58 @@ contract RFQTest is StrategySharedSetup {
+        bytes memory takerSig = _signRFQOrder(takerPrivateKey, newRFQOrder, SignatureValidator.SignatureType.EIP712);
 
         vm.expectRevert("invalid fee factor");
         bytes memory payload = _genFillRFQPayload(newRFQOrder, defaultMakerSig, defaultPermit, takerSig, defaultPermit);
         vm.prank(defaultOffer.taker, defaultOffer.taker);
         userProxy.toRFQv2(payload);
     }
 
-    function _signOffer(uint256 _privateKey, Offer memory _offer) private view returns (bytes memory sig) {
+    function _signOffer(
+        uint256 _privateKey,
+        Offer memory _offer,
+        SignatureValidator.SignatureType sigType
+    ) private view returns (bytes memory sig) {
         bytes32 offerHash = getOfferHash(_offer);
         bytes32 EIP712SignDigest = getEIP712Hash(rfq.EIP712_DOMAIN_SEPARATOR(), offerHash);
-        (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, EIP712SignDigest);
-        return abi.encodePacked(r, s, v);
+
+        if (
+            sigType == SignatureValidator.SignatureType.EIP712 ||
+            sigType == SignatureValidator.SignatureType.WalletBytes ||
+            sigType == SignatureValidator.SignatureType.WalletBytes32
+        ) {
+            (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, EIP712SignDigest);
+            sig = abi.encodePacked(r, s, v, uint8(sigType));
+        } else if (sigType == SignatureValidator.SignatureType.Wallet) {
+            (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, ECDSA.toEthSignedMessageHash(EIP712SignDigest));
+            sig = abi.encodePacked(r, s, v, uint8(sigType));
+        } else {
+            revert("Invalid signature type");
+        }
     }
 
-    function _signRFQOrder(uint256 _privateKey, RFQOrder memory _rfqOrder) private view returns (bytes memory sig) {
+    function _signRFQOrder(
+        uint256 _privateKey,
+        RFQOrder memory _rfqOrder,
+        SignatureValidator.SignatureType sigType
+    ) private view returns (bytes memory sig) {
         (, bytes32 rfqOrderHash) = getRFQOrderHash(_rfqOrder);
         bytes32 EIP712SignDigest = getEIP712Hash(rfq.EIP712_DOMAIN_SEPARATOR(), rfqOrderHash);
-        (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, EIP712SignDigest);
-        return abi.encodePacked(r, s, v);
+
+        if (
+            sigType == SignatureValidator.SignatureType.EIP712 ||
+            sigType == SignatureValidator.SignatureType.WalletBytes ||
+            sigType == SignatureValidator.SignatureType.WalletBytes32
+        ) {
+            (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, EIP712SignDigest);
+            sig = abi.encodePacked(r, s, v, uint8(sigType));
+        } else if (sigType == SignatureValidator.SignatureType.Wallet) {
+            (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, ECDSA.toEthSignedMessageHash(EIP712SignDigest));
+            sig = abi.encodePacked(r, s, v, uint8(sigType));
+        } else {
+            revert("Invalid signature type");
+        }
```

Comment:
nit: This part is kind of duplicate to those in `_signOffer`. Maybe we could extract a common `_signDigest(uint256, bytes, SignatureType)` to get rid of the duplicate.

Reply:
Sounds good. Fixed in cf380bc.